### PR TITLE
[new-parser] Avoid error when there is no space between rule keyword and rule name

### LINKED ISSUE #5895: [new-parser] No whitespace between rule keyword and rule name causes problems
## Parent issue
- #5678 

## Failing tests
- `org.drools.testcoverage.functional.parser.DrlParserTest#testParserSmoke`
  - `drools-test-coverage/test-suite/src/test/resources/org/drools/testcoverage/functional/parser/drl/whitespaces.drl`

## Rule code snippet
```
rule "5" when str :String() then end

rule"6" when then end rule"7" when then end
```

`LexerHelper` does not recognize `rule` in `rule"6"` as a valid end of the previous rule's RHS and so `rule"6"` is tokenized as `RHS_CHUNK`.

## Error output
```
### parse : ANTLR4_PARSER_ENABLED = true
00:26:32.836 [main] ERROR o.d.c.k.b.impl.AbstractKieProject.buildKnowledgePackages:280 - Unable to build KieBaseModel:KieBaseModelName
InvalidExpressionErrorResult: Unable to parse consequence caused by: (line 4,col 1) Parse error. Found  "\"6\"" <STRING_LITERAL>, expected one of  "," ";" "=" "@" "["
Problem stacktrace : 
  com.github.javaparser.GeneratedJavaParser.generateParseException(GeneratedJavaParser.java:13933)
  com.github.javaparser.GeneratedJavaParser.jj_consume_token(GeneratedJavaParser.java:13778)
  com.github.javaparser.GeneratedJavaParser.BlockStatement(GeneratedJavaParser.java:5807)
  com.github.javaparser.GeneratedJavaParser.Statements(GeneratedJavaParser.java:2773)
  com.github.javaparser.GeneratedJavaParser.Block(GeneratedJavaParser.java:5744)
  com.github.javaparser.GeneratedJavaParser.BlockParseStart(GeneratedJavaParser.java:8238)
  com.github.javaparser.JavaParser.parse(JavaParser.java:125)
  com.github.javaparser.JavaParser.parseBlock(JavaParser.java:317)
  org.drools.model.codegen.execmodel.generator.DrlxParseUtil.parseBlock(DrlxParseUtil.java:500)
  org.drools.model.codegen.execmodel.generator.Consequence.rewriteConsequence(Consequence.java:229)
  org.drools.model.codegen.execmodel.generator.Consequence.createCall(Consequence.java:143)
  org.drools.model.codegen.execmodel.generator.ModelGenerator.processRule(ModelGenerator.java:242)
  org.drools.model.codegen.execmodel.generator.ModelGenerator.processRuleDescr(ModelGenerator.java:185)
  org.drools.model.codegen.execmodel.generator.ModelGenerator.processRules(ModelGenerator.java:164)
  org.drools.model.codegen.execmodel.generator.ModelGenerator.generateModel(ModelGenerator.java:130)
  org.drools.model.codegen.execmodel.processors.ModelGeneratorPhase.process(ModelGeneratorPhase.java:48)
  org.drools.compiler.builder.impl.processors.IteratingPhase.process(IteratingPhase.java:53)
  org.drools.model.codegen.execmodel.processors.ModelMainCompilationPhase.process(ModelMainCompilationPhase.java:104)
  org.drools.model.codegen.execmodel.ModelBuilderImpl.doSecondBuildStep(ModelBuilderImpl.java:114)
  org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.build(CompositeKnowledgeBuilderImpl.java:125)
  org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.build(CompositeKnowledgeBuilderImpl.java:109)
  org.drools.compiler.kie.builder.impl.AbstractKieProject.buildKnowledgePackages(AbstractKieProject.java:274)
  org.drools.compiler.kie.builder.impl.AbstractKieProject.buildKnowledgePackages(AbstractKieProject.java:220)
  org.drools.compiler.kie.builder.impl.AbstractKieProject.verify(AbstractKieProject.java:84)
  org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildKieProject(KieBuilderImpl.java:285)
  org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildAll(KieBuilderImpl.java:251)
  org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildAll(KieBuilderImpl.java:208)
  org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem(KieUtil.java:147)
  org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem(KieUtil.java:134)
  org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromResources(KieUtil.java:121)
  org.drools.testcoverage.functional.parser.DrlParserTest.testParserSmoke(DrlParserTest.java:45)


00:26:32.842 [main] ERROR o.d.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem:165 - KieBuilder errors: [Message [id=1, kieBase=KieBaseModelName, level=ERROR, path=/var/home/jlocker/src/github.com/kiegroup/drools/drools-test-coverage/test-suite/src/test/resources/org/drools/testcoverage/functional/parser/drl/whitespaces.drl, line=35, column=0
   text=Unable to parse consequence caused by: (line 4,col 1) Parse error. Found  "\"6\"" <STRING_LITERAL>, expected one of  "," ";" "=" "@" "["
Problem stacktrace : 
  com.github.javaparser.GeneratedJavaParser.generateParseException(GeneratedJavaParser.java:13933)
  com.github.javaparser.GeneratedJavaParser.jj_consume_token(GeneratedJavaParser.java:13778)
  com.github.javaparser.GeneratedJavaParser.BlockStatement(GeneratedJavaParser.java:5807)
  com.github.javaparser.GeneratedJavaParser.Statements(GeneratedJavaParser.java:2773)
  com.github.javaparser.GeneratedJavaParser.Block(GeneratedJavaParser.java:5744)
  com.github.javaparser.GeneratedJavaParser.BlockParseStart(GeneratedJavaParser.java:8238)
  com.github.javaparser.JavaParser.parse(JavaParser.java:125)
  com.github.javaparser.JavaParser.parseBlock(JavaParser.java:317)
  org.drools.model.codegen.execmodel.generator.DrlxParseUtil.parseBlock(DrlxParseUtil.java:500)
  org.drools.model.codegen.execmodel.generator.Consequence.rewriteConsequence(Consequence.java:229)
  org.drools.model.codegen.execmodel.generator.Consequence.createCall(Consequence.java:143)
  org.drools.model.codegen.execmodel.generator.ModelGenerator.processRule(ModelGenerator.java:242)
  org.drools.model.codegen.execmodel.generator.ModelGenerator.processRuleDescr(ModelGenerator.java:185)
  org.drools.model.codegen.execmodel.generator.ModelGenerator.processRules(ModelGenerator.java:164)
  org.drools.model.codegen.execmodel.generator.ModelGenerator.generateModel(ModelGenerator.java:130)
  org.drools.model.codegen.execmodel.processors.ModelGeneratorPhase.process(ModelGeneratorPhase.java:48)
  org.drools.compiler.builder.impl.processors.IteratingPhase.process(IteratingPhase.java:53)
  org.drools.model.codegen.execmodel.processors.ModelMainCompilationPhase.process(ModelMainCompilationPhase.java:104)
  org.drools.model.codegen.execmodel.ModelBuilderImpl.doSecondBuildStep(ModelBuilderImpl.java:114)
  org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.build(CompositeKnowledgeBuilderImpl.java:125)
  org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.build(CompositeKnowledgeBuilderImpl.java:109)
  org.drools.compiler.kie.builder.impl.AbstractKieProject.buildKnowledgePackages(AbstractKieProject.java:274)
  org.drools.compiler.kie.builder.impl.AbstractKieProject.buildKnowledgePackages(AbstractKieProject.java:220)
  org.drools.compiler.kie.builder.impl.AbstractKieProject.verify(AbstractKieProject.java:84)
  org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildKieProject(KieBuilderImpl.java:285)
  org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildAll(KieBuilderImpl.java:251)
  org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildAll(KieBuilderImpl.java:208)
  org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem(KieUtil.java:147)
  org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem(KieUtil.java:134)
  org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromResources(KieUtil.java:121)
  org.drools.testcoverage.functional.parser.DrlParserTest.testParserSmoke(DrlParserTest.java:45)
]]

java.lang.AssertionError: [[Message [id=1, kieBase=KieBaseModelName, level=ERROR, path=/var/home/jlocker/src/github.com/kiegroup/drools/drools-test-coverage/test-suite/src/test/resources/org/drools/testcoverage/functional/parser/drl/whitespaces.drl, line=35, column=0
   text=Unable to parse consequence caused by: (line 4,col 1) Parse error. Found  "\"6\"" <STRING_LITERAL>, expected one of  "," ";" "=" "@" "["
Problem stacktrace : 
  com.github.javaparser.GeneratedJavaParser.generateParseException(GeneratedJavaParser.java:13933)
  com.github.javaparser.GeneratedJavaParser.jj_consume_token(GeneratedJavaParser.java:13778)
  com.github.javaparser.GeneratedJavaParser.BlockStatement(GeneratedJavaParser.java:5807)
  com.github.javaparser.GeneratedJavaParser.Statements(GeneratedJavaParser.java:2773)
  com.github.javaparser.GeneratedJavaParser.Block(GeneratedJavaParser.java:5744)
  com.github.javaparser.GeneratedJavaParser.BlockParseStart(GeneratedJavaParser.java:8238)
  com.github.javaparser.JavaParser.parse(JavaParser.java:125)
  com.github.javaparser.JavaParser.parseBlock(JavaParser.java:317)
  org.drools.model.codegen.execmodel.generator.DrlxParseUtil.parseBlock(DrlxParseUtil.java:500)
  org.drools.model.codegen.execmodel.generator.Consequence.rewriteConsequence(Consequence.java:229)
  org.drools.model.codegen.execmodel.generator.Consequence.createCall(Consequence.java:143)
  org.drools.model.codegen.execmodel.generator.ModelGenerator.processRule(ModelGenerator.java:242)
  org.drools.model.codegen.execmodel.generator.ModelGenerator.processRuleDescr(ModelGenerator.java:185)
  org.drools.model.codegen.execmodel.generator.ModelGenerator.processRules(ModelGenerator.java:164)
  org.drools.model.codegen.execmodel.generator.ModelGenerator.generateModel(ModelGenerator.java:130)
  org.drools.model.codegen.execmodel.processors.ModelGeneratorPhase.process(ModelGeneratorPhase.java:48)
  org.drools.compiler.builder.impl.processors.IteratingPhase.process(IteratingPhase.java:53)
  org.drools.model.codegen.execmodel.processors.ModelMainCompilationPhase.process(ModelMainCompilationPhase.java:104)
  org.drools.model.codegen.execmodel.ModelBuilderImpl.doSecondBuildStep(ModelBuilderImpl.java:114)
  org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.build(CompositeKnowledgeBuilderImpl.java:125)
  org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.build(CompositeKnowledgeBuilderImpl.java:109)
  org.drools.compiler.kie.builder.impl.AbstractKieProject.buildKnowledgePackages(AbstractKieProject.java:274)
  org.drools.compiler.kie.builder.impl.AbstractKieProject.buildKnowledgePackages(AbstractKieProject.java:220)
  org.drools.compiler.kie.builder.impl.AbstractKieProject.verify(AbstractKieProject.java:84)
  org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildKieProject(KieBuilderImpl.java:285)
  org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildAll(KieBuilderImpl.java:251)
  org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildAll(KieBuilderImpl.java:208)
  org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem(KieUtil.java:147)
  org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem(KieUtil.java:134)
  org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromResources(KieUtil.java:121)
  org.drools.testcoverage.functional.parser.DrlParserTest.testParserSmoke(DrlParserTest.java:45)
]]] 
Expecting empty but was: [Message [id=1, kieBase=KieBaseModelName, level=ERROR, path=/var/home/jlocker/src/github.com/kiegroup/drools/drools-test-coverage/test-suite/src/test/resources/org/drools/testcoverage/functional/parser/drl/whitespaces.drl, line=35, column=0
   text=Unable to parse consequence caused by: (line 4,col 1) Parse error. Found  "\"6\"" <STRING_LITERAL>, expected one of  "," ";" "=" "@" "["
Problem stacktrace : 
  com.github.javaparser.GeneratedJavaParser.generateParseException(GeneratedJavaParser.java:13933)
  com.github.javaparser.GeneratedJavaParser.jj_consume_token(GeneratedJavaParser.java:13778)
  com.github.javaparser.GeneratedJavaParser.BlockStatement(GeneratedJavaParser.java:5807)
  com.github.javaparser.GeneratedJavaParser.Statements(GeneratedJavaParser.java:2773)
  com.github.javaparser.GeneratedJavaParser.Block(GeneratedJavaParser.java:5744)
  com.github.javaparser.GeneratedJavaParser.BlockParseStart(GeneratedJavaParser.java:8238)
  com.github.javaparser.JavaParser.parse(JavaParser.java:125)
  com.github.javaparser.JavaParser.parseBlock(JavaParser.java:317)
  org.drools.model.codegen.execmodel.generator.DrlxParseUtil.parseBlock(DrlxParseUtil.java:500)
  org.drools.model.codegen.execmodel.generator.Consequence.rewriteConsequence(Consequence.java:229)
  org.drools.model.codegen.execmodel.generator.Consequence.createCall(Consequence.java:143)
  org.drools.model.codegen.execmodel.generator.ModelGenerator.processRule(ModelGenerator.java:242)
  org.drools.model.codegen.execmodel.generator.ModelGenerator.processRuleDescr(ModelGenerator.java:185)
  org.drools.model.codegen.execmodel.generator.ModelGenerator.processRules(ModelGenerator.java:164)
  org.drools.model.codegen.execmodel.generator.ModelGenerator.generateModel(ModelGenerator.java:130)
  org.drools.model.codegen.execmodel.processors.ModelGeneratorPhase.process(ModelGeneratorPhase.java:48)
  org.drools.compiler.builder.impl.processors.IteratingPhase.process(IteratingPhase.java:53)
  org.drools.model.codegen.execmodel.processors.ModelMainCompilationPhase.process(ModelMainCompilationPhase.java:104)
  org.drools.model.codegen.execmodel.ModelBuilderImpl.doSecondBuildStep(ModelBuilderImpl.java:114)
  org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.build(CompositeKnowledgeBuilderImpl.java:125)
  org.drools.compiler.builder.impl.CompositeKnowledgeBuilderImpl.build(CompositeKnowledgeBuilderImpl.java:109)
  org.drools.compiler.kie.builder.impl.AbstractKieProject.buildKnowledgePackages(AbstractKieProject.java:274)
  org.drools.compiler.kie.builder.impl.AbstractKieProject.buildKnowledgePackages(AbstractKieProject.java:220)
  org.drools.compiler.kie.builder.impl.AbstractKieProject.verify(AbstractKieProject.java:84)
  org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildKieProject(KieBuilderImpl.java:285)
  org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildAll(KieBuilderImpl.java:251)
  org.drools.compiler.kie.builder.impl.KieBuilderImpl.buildAll(KieBuilderImpl.java:208)
  org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem(KieUtil.java:147)
  org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem(KieUtil.java:134)
  org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromResources(KieUtil.java:121)
  org.drools.testcoverage.functional.parser.DrlParserTest.testParserSmoke(DrlParserTest.java:45)
]]

	at org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem(KieUtil.java:169)
	at org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromKieFileSystem(KieUtil.java:134)
	at org.drools.testcoverage.common.util.KieUtil.getKieBuilderFromResources(KieUtil.java:121)
	at org.drools.testcoverage.functional.parser.DrlParserTest.testParserSmoke(DrlParserTest.java:45)
```

### PR DESCRIPTION
**Pull request by yurloc**
Another corner case.

- Fixes #5895.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request and downstream checks</b>  
  - Push a new commit to the PR. An empty commit would be enough.

- for a <b>full downstream build</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- for <b>Jenkins PR check only</b>
  - If you are an ASF committer for KIE podling, login to Jenkins (https://ci-builds.apache.org/job/KIE/job/drools/), go to the specific PR job, and click on `Build Now` button.
</details>

### CODE REVIEW

**Comment by yurloc** [comment]
@mariofusco @tkobayas @gitgabrio please review.

File: drools-drl/drools-drl-parser/src/main/java/org/drools/drl/parser/antlr4/LexerHelper.java [hunk 1]
```diff
@@ -29,9 +29,10 @@
  */
 public class LexerHelper {
 
-    private static final List<Character> semiAndWS = List.of(';', ' ', '\t', '\n', '\r');
-    private static final List<String> statementKeywordsList = List.of(ParserHelper.statementKeywords);
-    private static final List<String> attributeKeywordsList = List.of(DroolsSoftKeywords.SALIENCE,
+    private static final Set<Character> semiAndWS = Set.of(';', ' ', '\t', '\n', '\r');
+    private static final Set<Character> delimiters = Set.of(';', ' ', '\t', '\n', '\r', '"', '\'');
+    private static final Set<String> statementKeywordsList = Set.of(ParserHelper.statementKeywords);
+    private static final Set<String> attributeKeywordsList = Set.of(DroolsSoftKeywords.SALIENCE,
```
Comment (yurloc) [inline drools-drl/drools-drl-parser/src/main/java/org/drools/drl/parser/antlr4/LexerHelper.java:35]: Not directly related to the fix but I thought it makes more sense to use a set here as the elements are unique, order is insignificant, and hash sets should have a faster lookup (_O(1)_).

File: drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/functional/parser/DrlParserTest.java [hunk 2]
```diff
@@ -34,7 +34,7 @@ public DrlParserTest(final File file, final KieBaseTestConfiguration kieBaseTest
         super(file, kieBaseTestConfiguration);
     }
 
-    @Parameters
+    @Parameters(name = "{index}: {0}, {1}")
```
Comment (yurloc) [inline drools-test-coverage/test-suite/src/test/java/org/drools/testcoverage/functional/parser/DrlParserTest.java:37]: Helps debugging as it's now clear which file is being tested.

File: drools-drl/drools-drl-parser/src/main/java/org/drools/drl/parser/util/ParserStringUtils.java [hunk 3]
```diff
@@ -33,7 +33,10 @@ private ParserStringUtils() {
     public static String safeStripStringDelimiters(String value) {
         if (value != null) {
             value = value.trim();
-            if (value.length() >= 2 && value.startsWith("\"") && value.endsWith("\"")) {
+            if (value.length() >= 2 && (
+                    value.startsWith("\"") && value.endsWith("\"")
+                    || value.startsWith("'") && value.endsWith("'")
+            )) {
```
Comment (yurloc) [inline drools-drl/drools-drl-parser/src/main/java/org/drools/drl/parser/util/ParserStringUtils.java:39]: Support single-quoted strings.

**Review by tkobayas — APPROVED** [review]

**Review by mariofusco — APPROVED** [review]

**Pull request merged**